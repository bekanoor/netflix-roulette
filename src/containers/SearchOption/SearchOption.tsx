interface IProps {
  searchTypeOnChange: (value: string) => void
}

const SearchOption = (props: IProps) => {
  const switchColor = () => {
    const buttons = document.querySelectorAll('.option-list__button')
    const firstBtn = buttons[0]
    const secondBtn = buttons[1]

    const condition = firstBtn.classList.contains('option-list__button--active')

    if (condition) {
      firstBtn.classList.remove('option-list__button--active')
      secondBtn.classList.add('option-list__button--active')
    } else {
      secondBtn.classList.remove('option-list__button--active')
      firstBtn.classList.add('option-list__button--active')
    }
  }

  const handleChangeTitle = () => {
    switchColor()
    props.searchTypeOnChange('title')
  }

  const handleChangeGenre = () => {
    switchColor()
    props.searchTypeOnChange('genre')
  }

  return (
    <nav>
      <ul className='option-list'>
        <li className='option-list__item'>
          <p>search by</p>
        </li>
        <li className='option-list__item'>
          <button
            onClick={handleChangeTitle}
            className='option-list__button option-list__button--active'
          >
            title
          </button>
        </li>
        <li className='option-list__item'>
          <button onClick={handleChangeGenre} className='option-list__button'>
            genre
          </button>
        </li>
      </ul>
    </nav>
  )
}

export { SearchOption }
