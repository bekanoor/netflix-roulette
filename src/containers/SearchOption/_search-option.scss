$primary-red: #F65261;

.option-list {
  display: flex;
  width: 300px;
  justify-content: space-between;
  align-items: center;

  &__item {
    color: white;
    text-transform: uppercase;
  }
  &__button {
    text-transform: uppercase;
    background-color: #555555;
    width: 80px;
    height: 30px;
    color: white;
    border-radius: 5px;
  }
  &__button--active {
    background-color: $primary-red;
  }
}