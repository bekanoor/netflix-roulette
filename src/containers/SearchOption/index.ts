import { SearchOption } from "./SearchOption";

export { SearchOption };