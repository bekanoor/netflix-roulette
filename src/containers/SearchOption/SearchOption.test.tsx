import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import { SearchOption } from './SearchOption'

describe('test search option component', () => {
  test('should render search option correctly', () => {
    const {container} = render(<SearchOption typeSwitcher={()=>{}}/>)

    expect(container).toMatchSnapshot()
  })

  test('dom elements really exist', () => {
    render(<SearchOption typeSwitcher={()=>{}}/>)

    expect(screen.getByTestId('search-option-test')).toBeInTheDocument
  }) 
})