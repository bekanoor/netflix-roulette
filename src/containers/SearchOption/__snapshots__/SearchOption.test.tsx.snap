// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test search option component should render search option correctly 1`] = `
<div>
  <nav
    data-testid="search-option-test"
  >
    <ul
      class="option-list"
    >
      <li
        class="option-list__item"
      >
        <p>
          search by
        </p>
      </li>
      <li
        class="option-list__item"
      >
        <button
          class="option-list__button option-list__button--active"
        >
          title
        </button>
      </li>
      <li
        class="option-list__item"
      >
        <button
          class="option-list__button"
        >
          genre
        </button>
      </li>
    </ul>
  </nav>
</div>
`;
